src/platform/packages/shared/presentation/presentation_publishing/interfaces/fetch/fetch.ts

```diff
@@ -149,13 +149,20 @@ export function fetch$(api: unknown): Observable<FetchContext> {
     }))
   );
 
+  const searchSessionId$ =
+    apiHasParentApi(api) && apiPublishesSearchSession(api.parentApi)
+      ? api.parentApi.searchSessionId$
+      : new BehaviorSubject(undefined);
+
   return merge(fetchContext$, reload$).pipe(
-    combineLatestWith(isFetchPaused$),
-    filter(([, isFetchPaused]) => !isFetchPaused),
-    map(([fetchContext]) => fetchContext as ReloadTimeFetchContext),
-    distinctUntilChanged((prevContext, nextContext) =>
-      isReloadTimeFetchContextEqual(prevContext, nextContext)
+    combineLatestWith(searchSessionId$, isFetchPaused$),
+    filter(([, , isFetchPaused]) => !isFetchPaused),
+    distinctUntilChanged(
+      ([prevContext, prevSearchSessionId], [nextContext, nextSearchSessionId]) =>
+        isReloadTimeFetchContextEqual(prevContext, nextContext) &&
+        prevSearchSessionId === nextSearchSessionId
```

Comment:
We actually don't want to avoid fetching when search session IDs are equal. Otherwise, we do not re-fetch when selections are made on a control that lives in a collapsible section. See https://github.com/elastic/kibana/pull/240333 and https://github.com/elastic/kibana/pull/239221